### ISSUE TITLE
multi-card（intel-gpu） training 只有workerlog.0有数据，其他workerlog就几行log

### ISSUE THREAD

**Issue by chaofanl** [PaddlePaddle/PaddleClas#2914]
1. PaddleClas版本是develop
2. 训练环境：
    a. linux22.04
    b.python3.10
    c.intel-gpu

workerlog.0 log :
[2023/08/16 19:54:43] ppcls INFO: profiler_options : None
[2023/08/16 19:54:43] ppcls INFO: train with paddle 0.0.0 and device Place(intel_gpu:0)
[2023/08/16 19:54:45] ppcls WARNING: "init_res" will be deprecated, please use "init_net" instead.
[2023/08/16 19:54:45] ppcls WARNING: When using warm up, the value of "Global.epochs" must be greater than value of "Optimizer.lr.warmup_epoch". The value of "Optimizer.lr.warmup_epoch" has been set to 5.
[2023/08/16 19:54:45] ppcls WARNING: The training strategy provided by PaddleClas is based on 4 gpus. But the number of gpu is 2 in current training. Please modify the stategy (learning rate, batch size and so on) if use this config to train.
I0816 19:54:45.905280 1947284 tcp_utils.cc:181] The server starts to listen on IP_ANY:41848
I0816 19:54:45.905407 1947284 tcp_utils.cc:130] Successfully connected to 10.165.58.187:41848
2023:08:16-19:54:45:(1947284) |CCL_WARN| could not get local_idx/count from environment variables, trying to get them from ATL
2023:08:16-19:54:45:(1947284) |CCL_WARN| sockets exchange mode is set. It may cause potential problem of 'Too many open file descriptors'
[2023/08/16 19:54:48] ppcls INFO: Set random seed to (0 + $PADDLE_TRAINER_ID) for different trainer
[2023/08/16 19:55:18] ppcls INFO: [Train][Epoch 1/5][Iter: 0/14]lr(LinearWarmup): 0.00017857, top1: 0.00000, top5: 0.05859, CELoss: 4.83407, loss: 4.83407, hw_cost: 28.08635s, batch_cost: 30.07965s, reader_cost: 1.98999, hw: 9.11475 fps, ips: 8.51074 samples/s, eta: 0:35:05
[2023/08/16 19:55:18] ppcls INFO: [Train][Epoch 1/5][Iter: 1/14]lr(LinearWarmup): 0.00035714, top1: 0.00000, top5: 0.07031, CELoss: 4.83559, loss: 4.83559, hw_cost: 14.43727s, batch_cost: 15.43543s, reader_cost: 0.99506, hw: 324.79581 fps, ips: 16.58522 samples/s, eta: 0:17:45
[2023/08/16 19:55:19] ppcls INFO: [Train][Epoch 1/5][Iter: 2/14]lr(LinearWarmup): 0.00053571, top1: 0.00000, top5: 0.04688, CELoss: 4.89722, loss: 4.89722, hw_cost: 9.88877s, batch_cost: 10.55530s, reader_cost: 0.66340, hw: 323.32837 fps, ips: 24.25322 samples/s, eta: 0:11:57
[2023/08/16 19:55:20] ppcls INFO: [Train][Epoch 1/5][Iter: 3/14]lr(LinearWarmup): 0.00071429, top1: 0.00000, top5: 0.03516, CELoss: 4.95507, loss: 4.95507, hw_cost: 7.61374s, batch_cost: 8.11436s, reader_cost: 0.49756, hw: 324.60562 fps, ips: 31.54899 samples/s, eta: 0:09:03
[2023/08/16 19:55:21] ppcls INFO: [Train][Epoch 1/5][Iter: 4/14]lr(LinearWarmup): 0.00089286, top1: 0.01016, top5: 0.04766, CELoss: 4.84117, loss: 4.84117, hw_cost: 6.24948s, batch_cost: 6.65057s, reader_cost: 0.39806, hw: 323.04976 fps, ips: 38.49293 samples/s, eta: 0:07:18
[2023/08/16 19:55:22] ppcls INFO: [Train][Epoch 1/5][Iter: 5/14]lr(LinearWarmup): 0.00107143, top1: 0.00846, top5: 0.03971, CELoss: 4.93871, loss: 4.93871, hw_cost: 0.79408s, batch_cost: 0.79694s, reader_cost: 0.00009, hw: 322.38613 fps, ips: 321.23011 samples/s, eta: 0:00:51
[2023/08/16 19:55:22] ppcls INFO: [Train][Epoch 1/5][Iter: 6/14]lr(LinearWarmup): 0.00125000, top1: 0.00725, top5: 0.06027, CELoss: 4.92091, loss: 4.92091, hw_cost: 0.79583s, batch_cost: 0.79864s, reader_cost: 0.00008, hw: 320.97142 fps, ips: 320.54559 samples/s, eta: 0:00:51
[2023/08/16 19:55:23] ppcls INFO: [Train][Epoch 1/5][Iter: 7/14]lr(LinearWarmup): 0.00142857, top1: 0.00684, top5: 0.07080, CELoss: 4.88681, loss: 4.88681, hw_cost: 0.79366s, batch_cost: 0.79648s, reader_cost: 0.00007, hw: 324.32726 fps, ips: 321.41556 samples/s, eta: 0:00:50

workerlog.1 只有几行log
I0816 19:54:42.765476 1947285 init.cc:232] ENV [CUSTOM_DEVICE_ROOT]=/home/afas/miniconda3/envs/chaofanl/lib/python3.10/site-packages/paddle_custom_device
I0816 19:54:42.765515 1947285 init.cc:141] Try loading custom device libs from: [/home/afas/miniconda3/envs/chaofanl/lib/python3.10/site-packages/paddle_custom_device]
I0816 19:54:42.834534 1947285 custom_device.cc:1042] Successed in loading custom runtime in lib: /home/afas/miniconda3/envs/chaofanl/lib/python3.10/site-packages/paddle_custom_device/libpaddle-custom-intel-gpu.so
I0816 19:54:42.835247 1947285 custom_kernel.cc:76] Successed in loading 94 custom kernel(s) from loaded lib(s), will be used like native ones.
I0816 19:54:42.835278 1947285 init.cc:153] Finished in LoadCustomDevice with libs_path: [/home/afas/miniconda3/envs/chaofanl/lib/python3.10/site-packages/paddle_custom_device]
I0816 19:54:42.835289 1947285 init.cc:238] CustomDevice: intel_gpu, visible devices count: 1
I0816 19:54:45.908782 1947285 tcp_utils.cc:130] Successfully connected to 10.165.58.187:41848
2023:08:16-19:54:46:(1947285) |CCL_WARN| could not get local_idx/count from environment variables, trying to get them from ATL
2023:08:16-19:54:46:(1947285) |CCL_WARN| sockets exchange mode is set. It may cause potential problem of 'Too many open file descriptors'

**Comment by chaofanl** [comment]
run command
python -m paddle.distributed.launch --gpus "0,1" ./tools/train.py -c /home/user/ResNet50-dist.yaml

**Comment by TingquanGao** [comment]
这是正常的。只有0号卡的进程输出信息，如果需要保留每张卡的进程的输出，可以简单改下这里的代码：

https://github.com/PaddlePaddle/PaddleClas/blob/release/2.5/ppcls/utils/logger.py#L88-L103

**Comment by chaofanl** [comment]
谢谢你这么快的回复，那我如果算performance得话是不是workerlog.0的数据乘以卡的数量啊？

**Comment by TingquanGao** [comment]
嗯嗯，这样可以的

**Comment by chaofanl** [comment]
非常感谢，那我把这个issue 关掉了
